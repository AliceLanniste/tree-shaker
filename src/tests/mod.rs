fn tree_shake(input: &str) {
  let result = crate::tree_shake(input);
  let output = result.codegen_return.source_text;
  insta::assert_snapshot!(output);
}

#[test]
fn test_1() {
  tree_shake(
    r#"
      export let a = 1 && 2;
      export { b };
      let c = 3;
      let { ["b"]: b, d } = { b: 2, d: effect };
      "#,
  );
}
