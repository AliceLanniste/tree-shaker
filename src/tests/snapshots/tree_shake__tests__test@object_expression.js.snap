---
source: src/tests/mod.rs
expression: tree_shake(&input)
input_file: src/tests/fixtures/object_expression.js
---
export function main() {
	effect(1, 3);
	let __unused_4CB3 = { unused: effect() };
	let __unused_0648 = { get b() {
		effect();
	} };
	effect(2);
	obj3.b;
}
